sdks/go/pkg/beam/core/runtime/exec/fn.go

```diff
@@ -261,21 +274,34 @@ func (n *invoker) ret2(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime,
 // ret3 handles processing of a trio of return values.
 func (n *invoker) ret3(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime, r0, r1, r2 interface{}) (*FullValue, error) {
 	switch {
-	case n.outErrIdx >= 0:
-		if r2 != nil {
-			return nil, r2.(error)
+	case n.outEtIdx >= 0:
+		if n.outErrIdx == 2 {
+			if r2 != nil {
+				return nil, r2.(error)
+			}
+			n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn}
+			return &n.ret, nil
 		}
-		if n.outEtIdx < 0 {
-			n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Elm2: r1, Pane: pn}
+		if n.outPcIdx >= 0 {
```

Comment:
There's a lot of mixing `>=` conditions and `==` conditions on how we check these indices - we should coallesce to one approach. As a reader, the back and forth makes it harder to follow. When possible, I'd vote to prefer the exact `==` check since validation of signatures should be taken care of on the front end, and if we don't have the indices right we'll struggle later on anyways.

Reply:
You're right, this could use some cleaning up. I think the major juggling comes down to "do we have a V or a KV?" and "do we have an error to check?" and those cases can get messy. But we should be able to do a little more with exact indices. Another thing to tweak

Reply:
Committed to explict `==` checks since we know what indices they should be at